packages/cadl-python/tasks.py

```diff
@@ -33,11 +32,37 @@ def regenerate(c, name=None, debug=False):
     specs = [s for s in specs if name.lower() in s.stem.lower()]
   for spec in specs:
     Path(f"{PLUGIN_DIR}/test/generated/{spec.name}").mkdir(parents=True, exist_ok=True)
+  return specs
+
+def _emitter_path():
+    return (PLUGIN_DIR / "dist/src/index.js").as_posix()
+
+def _execute(specs, debug):
   _run_cadl([
-    f"cadl compile {spec} --emit={PLUGIN_DIR}/dist/src/index.js --output-dir={PLUGIN_DIR}/test/generated/{spec.name}{' --debug' if debug else ''}"
+    f"cadl compile {spec} --emit={_emitter_path()} --output-dir={PLUGIN_DIR}/test/generated/{spec.name}{' --debug' if debug else ''}"
     for spec in specs
   ])
 
+@task
+def regenerate_cadl_project(c, name=None, debug=False):
+  specs = _all_specs(name)
+  specs = [s for s in specs if "hello" in s.name]
```

Comment:
this is too hacky and won't grow well. I'm wondering if instead of having a bunch of different `regenerate` functions, we could have a section where each test can specify their options, so we can always generate all of them with the options they need

Reply:
@iscai-msft updated